### PR TITLE
raise auto piece size selection limit to 16 MB in create_torrent() (master branch)

### PR DESCRIPTION
**Pull request by Chocobo1**
Same as #2669, but for master branch.

### CODE REVIEW

File: src/create_torrent.cpp [hunk 1]
```diff
@@ -328,16 +328,20 @@ namespace {
...
-			piece_size = int(fs.total_size() / (target_size / 20));
-
-			int i = 16*1024;
-			for (; i < 2*1024*1024; i *= 2)
+			// size_table is computed from the following:
+			//   target_list_size = sqrt(total_size) * 2;
+			//   target_piece_size = total_size / (target_list_size / hash_size);
+			// Given hash_size = 20 bytes, target_piece_size = (16*1024 * pow(2, i))
+			// we can determine size_table = (total_size = pow(2 * target_piece_size / hash_size, 2))
+			boost::int64_t const size_table[] = {2684355, 10737418, 42949673, 171798692, 687194767,
```
Comment (Chocobo1) [inline src/create_torrent.cpp]: an alternative for this may be:
```c++
std::array<int64_t> const size_table {/*values*/};
```

**Comment by arvidn** [comment]
this is the same patch as for ``RC_1_1``, right?

**Comment by Chocobo1** [comment]
>this is the same patch as for RC_1_1, right?

Yes. Just a convenient for you to pull in.

**Comment by arvidn** [comment]
I prefer to merge it from RC_1_1

**Comment by Chocobo1** [comment]
>I prefer to merge it from RC_1_1

I understand, closing.

**Pull request closed without merging**